Frameworks/UIKit/UITextField.mm

```diff
@@ -120,34 +113,30 @@ @implementation UITextField {
  @Status Interoperable
 */
 - (void)setText:(NSString*)text {
-    if (![_text isEqualToString:text]) {
-        if (text != nil) {
-            _text = [text copy];
-        } else {
-            _text = nil;
-        }
-
-        [_secureModeLock lock];
-        if (_secureTextMode) {
-            _passwordBox.password = _text;
-        } else {
-            _textBox.text = _text;
-            // Ensure caret at end of field in case we programmatically
-            // gain focus (becomeFirstResponder) after the text is set:
-            _textBox.selectionStart = [_text length];
-            _textBox.selectionLength = 0;
-        }
-        [_secureModeLock unlock];
-
-        [self _adjustFontSizeToFitWidthOrApplyCurrentFont];
+    [_secureModeLock lock];
+    if (_secureTextMode) {
+        _passwordBox.password = [text copy];
+    } else {
+        _textBox.text = [text copy];
+        // Ensure caret at end of field in case we programmatically
+        // gain focus (becomeFirstResponder) after the text is set:
+        _textBox.selectionStart = [text length];
+        _textBox.selectionLength = 0;
     }
+    [_secureModeLock unlock];
+
+    [self _adjustFontSizeToFitWidthOrApplyCurrentFont];
 }
 
 /**
  @Status Interoperable
 */
 - (NSString*)text {
```

Comment:
@tadam-msft, should we add some UITextFiled tests that set/retrieve some of these values? or are we still waiting to finalize a pattern?  I think the pattern's relatively clear at this point.